tests/unit/test.program.js

```diff
@@ -480,21 +507,154 @@ describe('program.main', () => {
       return configObject;
     });
 
-    return execProgram(
+    await execProgram(
       ['lint', '--config', 'path/to/web-ext-config.js'],
       {
         commands: fakeCommands,
         runOptions: {
           loadJSConfigFile: fakeLoadJSConfigFile,
         },
-      })
-      .then(() => {
-        const options = fakeCommands.lint.firstCall.args[0];
-        // This makes sure that the config object was applied
-        // to the lint command options.
-        assert.equal(
-          options.selfHosted, configObject.lint.selfHosted);
-      });
+      }
+    );
+
+    const options = fakeCommands.lint.firstCall.args[0];
+    // This makes sure that the config object was applied
+    // to the lint command options.
+    assert.equal(
+      options.selfHosted, configObject.lint.selfHosted);
+  });
+
+  it('discovers config files', async () => {
+    const fakeCommands = fake(commands, {
+      lint: () => Promise.resolve(),
+    });
+    const configObject = {
+      lint: {
+        selfHosted: true,
+      },
+    };
+    // Instead of loading/parsing a real file, just return an object.
+    const fakeLoadJSConfigFile = sinon.spy(() => {
+      return configObject;
+    });
+
+    await execProgram(
+      ['lint'],
+      {
+        commands: fakeCommands,
+        runOptions: {
+          discoverConfigFiles: async () => ['fake/config.js'],
```

Comment:
we could also wrap this `discoverConfigFiles` into a sinon.spy and check that it has been called as expected.

Reply:
I prefer to check things like that implicitly when possible. For this test case, if `discoverConfigFiles` is not called, you get this error:

```
  1) program.main discovers config files:
     AssertionError: expected false to equal true
      at _callee5$ (dist/webpack:/tests/unit/test.program.js:556:12)
      at tryCatch (node_modules/regenerator-runtime/runtime.js:62:40)
      at GeneratorFunctionPrototype.invoke [as _invoke] (node_modules/regenerator-runtime/runtime.js:296:22)
      at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (node_modules/regenerator-runtime/runtime.js:114:21)
      at step (dist/webpack:/node_modules/babel-runtime/helpers/asyncToGenerator.js:17:1)
      at dist/webpack:/node_modules/babel-runtime/helpers/asyncToGenerator.js:28:1
```

When you look up the line of the assertion failure you will see this:

```js
    // This makes sure that the config object was applied
    // to the lint command options.
    assert.equal(
      options.selfHosted, configObject.lint.selfHosted);
```

I think that will be helpful to diagnose the problem.